### PR TITLE
Implemented "add_unique_key" feature for Cosmos DB cosmosContainer.

### PR DESCRIPTION
**Pull request by JordanMarr**

### CODE REVIEW

File: src/Farmer/Arm/DocumentDb.fs [hunk 1]
```diff
@@ -36,6 +38,10 @@ module DatabaseAccounts =
                                   partitionKey =
                                    {| paths = this.PartitionKey.Paths
                                       kind = string this.PartitionKey.Kind |}
+                                  uniqueKeyPolicy = 
+                                   {| uniqueKeys = 
+                                      this.UniqueKeyPolicy.UniqueKeys 
+                                      |> List.map (fun k -> {| paths = k.Paths |}) |}
```
Comment (isaacabraham) [inline src/Farmer/Arm/DocumentDb.fs:44]: Can this be simplified to just `uniqueKeyPolicy = this.UniqueKeyPolicy`?

File: src/Farmer/Builders/Builders.Cosmos.fs [hunk 2]
```diff
@@ -73,6 +81,7 @@ type CosmosDbContainerBuilder() =
         { Name = ResourceName ""
           PartitionKey = [], Hash
           Indexes = []
+          UniqueKeys = []
```
Comment (isaacabraham) [inline src/Farmer/Builders/Builders.Cosmos.fs:84]: Should this be a set or dictionary (do we need to worry about uniqueness of paths or something)?

File: src/Farmer/Arm/DocumentDb.fs (hunk 1 above)
Comment (JordanMarr) [reply src/Farmer/Arm/DocumentDb.fs:44]: I was just manually mapping through the `this.UniqueKeyPolicy.UniqueKeys` to manually create camel cased versions of the children in the object graph (since there is an extra layer beneath that).  I agree that it's a little cumbersome compared to just using a camel case serialization in Newtonsoft.Json, but I was trying to keep it simple and just copy the methodology that was already in-place above.

Btw, here is the desired output:

image

File: src/Farmer/Builders/Builders.Cosmos.fs (hunk 2 above)
Comment (JordanMarr) [reply src/Farmer/Builders/Builders.Cosmos.fs:84]: That's a great point -- duplicate paths would likely be illegal.

File: src/Farmer/Arm/DocumentDb.fs (hunk 1 above)
Comment (JordanMarr) [reply src/Farmer/Arm/DocumentDb.fs:44]: And for good measure, here is what was generated if two unique keys are entered via Azure Portal (and what this currently generates as well):

image

**Comment by JordanMarr** [comment]
I changed UniqueKeys from a List to a Set.  Also added a Cosmos unit test to verify that errant duplicate keys are ignored.

**Comment by JordanMarr** [comment]
I was just doing some more testing on Azure which prompted to me to add an important hint to the "add_unique_key" property: the uniqueness of the key only applies within the logical partition.  If someone doesn't know this, they might think that there was an issue with the ARM template.  This comment will hopefully help with that.
